code/datums/storage/subtypes/rped.dm

```diff
@@ -9,7 +13,81 @@
 	max_specific_storage = WEIGHT_CLASS_NORMAL
 	numerical_stacking = TRUE
 
+	/**
+	 * as of now only these stack components are required to build machines like[thermomaachine,crystallizer,electrolyzer]
+	 * so we limit the rped to pick up only these stack types so players dont cheat and use this as a general storage medium
+	 */
+	var/static/list/allowed_material_types=list(
+		/obj/item/stack/sheet/glass,
+		/obj/item/stack/sheet/plasteel,
+		/obj/item/stack/cable_coil,
+	)
+
+	/**
+	 * we check if the user is trying to insert any of these bluespace crystal types into the RPED
+	 * at any point the total sum of all these types in the RPED must be 30
```

Comment:
Where does the number 30 come from? You should reference the variable instead of a flat number, as that might change

Reply:
It comes from the MAX_PICKUP_TYPE variable.  I'll update the comment